### PR TITLE
Drop maintaining own versions of dependencies

### PR DESCRIPTION
**Pull request by omerfirmak**
Prevents having mismatched versions of dependencies between faucet
and agora

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,63 +1,3 @@
...
-[submodule "submodules/crypto"]
-	path = submodules/crypto
-	url = https://github.com/bosagora/crypto.git
-[submodule "submodules/serialization"]
-	path = submodules/serialization
-	url = https://github.com/bosagora/serialization.git
-[submodule "config"]
-	path = submodules/config
-	url = https://github.com/bosagora/config.git
+	url = https://github.com/bosagora/agora.git
```
Comment (mkykadir) [inline .gitmodules:3]: new line

**Comment by mkykadir** [comment]
It's cool but we might also need submodules to be initialized inside `agora` due to nesting

File: .gitmodules [hunk 2]
```diff
@@ -1,63 +1,3 @@
 [submodule "agora"]
 	path = submodules/agora
```
Comment (mkykadir) [inline .gitmodules:2]: Check [this](https://git-scm.com/docs/gitmodules#Documentation/gitmodules.txt-submoduleltnamegtfetchRecurseSubmodules) for recursive submodule init

**Review by mkykadir — CHANGES_REQUESTED** [review]
It's cool! But we might also need submodules to be initialized inside agora due to nesting

**Comment by omerfirmak** [comment]
all green

**Review by mkykadir — APPROVED** [review]

**Comment by Geod24** [comment]
I usually avoid recursive submodules because it's generally a pain to manage. But in that instance, it is a much better solution. Thanks!

**Pull request merged**